src/main/java/CharArraySolution.java

```diff
@@ -1,44 +1,41 @@
 import java.util.Arrays;
 
 public class CharArraySolution {
-    int string_counter = 0;
 
     public int solveEquation(String equation) {
 
-        if (Utils.getInstance().verifyEquationInput(equation)) {
+        if (Utils.verifyEquationInput(equation)) {
             char[] characters = equation.toCharArray();
-            int operatorLocation = findOperator(characters);
+            int operator_location = findOperator(characters);
 
-            if (verifyOperator(Character.toString(characters[operatorLocation]), operatorLocation)) {
-                int numBefore = Integer.parseInt(
-                        String.valueOf(
-                                Arrays.copyOfRange(characters, 0, operatorLocation))
-                );
-                int numAfter = Integer.parseInt(
-                        String.valueOf(
-                                Arrays.copyOfRange(characters, operatorLocation + 1, characters.length))
-                );
-                return Utils.getInstance().calcRes(Character.toString(characters[operatorLocation]), numBefore, numAfter);
+            if (operator_location != -1) {
+                if (Utils.VALID_OPERATORS.contains(Character.toString(characters[operator_location]))) {
+                    int numBefore = Integer.parseInt(
+                            String.valueOf(Arrays.copyOfRange(characters, 0, operator_location)));
+                    int numAfter = Integer.parseInt(
+                            String.valueOf(Arrays.copyOfRange(characters, operator_location + 1, characters.length)));
+                    return Utils.solveEquation(Character.toString(characters[operator_location]), numBefore, numAfter);
+                } else {
+                    return -1;
+                }
             }
         }
         return -1;
     }
 
-    public int findOperator(char[] characters) {
+    private int findOperator(char[] characters) {
+        int string_counter = 0;
```

Comment:
in Java/Kotlin we use [camel case](https://en.wikipedia.org/wiki/Camel_case)

Reply:
done